cmd/bancheck/analyzers/banned_api_test.go

```diff
@@ -0,0 +1,242 @@
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package analyzers
+
+import (
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"golang.org/x/tools/go/analysis/analysistest"
+)
+
+func TestBannedAPIAnalyzer(t *testing.T) {
+	for _, test := range []struct {
+		desc  string
+		files map[string]string
+	}{
+		{
+			desc: "No banned APIs",
+			files: map[string]string{
+				"config.json": `
+				{}
+				`,
+				"main/test.go": `
+				package main;
+				func main() {}
+				`,
+			},
+		},
+		{
+			desc: "Banned APIs exist",
+			files: map[string]string{
+				"config.json": `
+				{
+					"functions": [
+						{
+							"name": "fmt.Printf",
+							"msg": "Banned by team A"
+						}
+					],
+					"imports": [
+						{
+							"name": "fmt",
+							"msg": "Banned by team A"
+						}
+					]
+				}
+				`,
+				"main/test.go": `
+				package main
+
+				import "fmt" // want "Banned API found \"fmt\". Additional info: Banned by team A"
+
+				func main() {
+					fmt.Printf("Hello") // want "Banned API found \"fmt.Printf\". Additional info: Banned by team A"
+				}
+				`,
+			},
+		},
+		{
+			desc: "Banned APIs in exempted package",
+			files: map[string]string{
+				"config.json": `
+				{
+					"functions": [
+						{
+							"name": "fmt.Printf",
+							"msg": "Banned by team A",
+							"exemptions": [
+								{
+									"justification": "#yolo",
+									"allowedPkg": "main"
+								}
+							]
+						}
+					],
+					"imports": [
+						{
+							"name": "fmt",
+							"msg": "Banned by team A",
+							"exemptions": [
+								{
+									"justification": "#yolo",
+									"allowedPkg": "main"
+								}
+							]
+						}
+					]
+				}
+				`,
+				"main/test.go": `
+				package main
+
+				import "fmt"
+
+				func main() {
+					fmt.Printf("Hello")
+				}
+				`,
+			},
+		},
+		{
+			desc: "Banned renamed import",
+			files: map[string]string{
+				"config.json": `
+				{
+					"imports": [
+						{
+							"name": "fmt",
+							"msg": "Banned by team A"
+						}
+					]
+				}
+				`,
+				"main/test.go": `
+				package main
+
+				import allowed "fmt" // want "Banned API found \"fmt\". Additional info: Banned by team A"
+
+				func main() {
+					allowed.Printf("Hello")
+				}
+				`,
+			},
+		},
+		{
+			desc: "Banned function from renamed import",
+			files: map[string]string{
+				"config.json": `
+				{
+					"functions": [
+						{
+							"name": "fmt.Printf",
+							"msg": "Banned by team A"
+						}
+					]
+				}
+				`,
+				"main/test.go": `
+				package main
+
+				import allowed "fmt"
```

Comment:
nit: `renamed`

I understand your intent here, but reading `allowed` here makes the test confusing.

Reply:
done